pkg/status/controller.go

```diff
@@ -328,8 +328,16 @@ func updateObjectStatus(ctx context.Context, objRef *util.SourceRef, status map[
 
 	if objRef.Namespace == "" {
 		_, err = wdsDynClient.Resource(gvr).UpdateStatus(ctx, unstrObj, metav1.UpdateOptions{})
+		// if resource not found it may mean no status subresource - try to update the whole resource
+		if errors.IsNotFound(err) {
+			_, err = wdsDynClient.Resource(gvr).Update(ctx, unstrObj, metav1.UpdateOptions{})
+		}
```

Comment:
This works, but may not be efficient as for non-subresources statuses there are alway two calls for each status update, the first of which triggers the not found error, and considering that status can be updated quite often may not be optimal. I would suggest to think about a solution that can lookup if status is a sub-resource or not and then does the correct thing. This was done previously in legacy KS - please have a look at https://github.com/kubestellar/kubestellar/pull/1228

Reply:
After looking at https://github.com/kubestellar/kubestellar/pull/1228, I don't think this will be more efficient, because the subresource lookup also has an API call under the hood in the `restmapper.GetAPIGroupResources(discoveryClient)` to get the `groupResources`. And we need to get the `groupResources` list for each status update, because if we do this once, then any CRD added after this point in time will not be on the list.
WDYT?

Reply:
@andreyod you are correct, if the restmapper.GetAPIGroupResources(discoveryClient) is called every time this would be even worse (especially if there are many api resources such as in OCP). I was suggesting that since we do have code in the binding controller where we detect new CRDs we could at that point update the rest mapper and/or related structs that would keep the flag about status subresource. However this might be a further optimization I am ok to defer, so for now I am fine with the current approach you implemented since it is simple and works for the current purpose.